velox/exec/Task.cpp

```diff
@@ -917,13 +923,17 @@ void Task::resume(std::shared_ptr<Task> self) {
             continue;
           }
           VELOX_CHECK(!driver->isOnThread() && !driver->isTerminated());
-          if (!driver->state().hasBlockingFuture) {
+          if (!driver->state().hasBlockingFuture &&
+              driver->task()->queryCtx()->isExecutorSupplied()) {
             if (driver->state().endExecTimeMs != 0) {
               driver->state().totalPauseTimeMs +=
                   getCurrentTimeMs() - driver->state().endExecTimeMs;
             }
             // Do not continue a Driver that is blocked on external
             // event. The Driver gets enqueued by the promise realization.
+            //
+            // Do not continue the driver if no executor is supplied,
+            // This usually happens in single-threaded execution.
```

Comment:
```
This usually happens in serialized task execution mode.
```

Reply:
As discussed, delay to subsequent PR.

Reply:
Hi @tanjialiang 

The file has other places mentioning `single-threaded execution` than `serial execution`. I'd suggest doing a clean up to minimize the usages of `single-threaded execution`. Before that we can keep this file consistent on the phasing.